test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -389,16 +347,12 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 							address := node.Status.GetAddressByType("InternalIP")
 							if address == nil {
 								log.Printf("One of our nodes does not have an InternalIP value!: %s\n", node.Metadata.Name)
+								Fail("All nodes should have an InternalIP address")
 							}
-							Expect(address).NotTo(Equal(nil))
```

Comment:
was this not failing when address was nil?

Reply:
I think the way ginkgo implements the equality check doesn't work in the way it was being used.

Reply:
looking at gingko docs it looks like `Fail()`  won't run the `AfterSuite()` stuff so maybe not ideal

http://onsi.github.io/ginkgo/#getting-started-writing-your-first-test

Reply:
could we maybe add an `else{}` to protect against panic instead and let the assertion do its thing?

Reply:
Good catch. What do you think about this `getAddressError` implementation?

Reply:
I don't think that will help. The succeeding line will still be run without an else statement to protect it. I think `Expect(address).NotTo(Equal(nil))` <==> `Expect(getAddressError).NotTo(HaveOccurred())`

Reply:
Should be `Expect(address).NotTo(BeNil())`